src/services/postService.js

```diff
@@ -53,9 +79,33 @@ async function checkLike(like, postID, userID){
     return postData;
 }
 
+const deletePost = async (postId) => {
+    await getPostById(postId);
+
+    const deleteResult = await postDAO.deletePost(postId);
+    throwIfError(deleteResult);
+}
+
+const deleteReply = async (postId, replyId) => {
+    const repliesOfPost = await getRepliesOfPost(postId);
+    
+    const index = repliesOfPost.findIndex((reply) => reply.itemID === replyId);
+    if (index === -1) {
+        throw { status: 400, message: "That reply doesn't exist" }
+    }
```

Comment:
Your getReplyOfPost function does everything above this line of code, but you don't use it. I assume its because you need the index and not the reply. I don't think there is a use case for a user to get a single reply from the server so you can probably just change it to return the index, but I could be wrong. It doesn't look like anything else uses that function so you can use it here or delete it.